File: libre-asi/src/lib/models/Question.ts
import type { Answer } from './Answer';
import type QuestionOption from './Option';

export interface Question {
	id?: number;
	special_id?: string;
	statement?: string;
	order?: number;
	type?: string;
	category?: string;
	options?: QuestionOption[];
	answers: Answer[];
	valid: boolean;
}

File: api/pkg/view/question.go
package view

type Question struct {
	ID        uint     `json:"id"`
	SpecialID string   `json:"special_id"`
	Statement string   `json:"statement"`
	Order     int      `json:"order"`
	Type      string   `json:"type"`
	Category  string   `json:"category"`
	Options   []Option `json:"options"`
	Answer    []Answer `json:"answer"`
}
